analyzer/javacg-opal/src/main/java/eu/fasten/analyzer/javacgopal/OPALPlugin.java

```diff
@@ -76,9 +79,18 @@ public void consume(String kafkaRecord, ProcessingLane l) {
                 // Generate CG and measure construction duration.
                 logger.info("[CG-GENERATION] [UNPROCESSED] [-1] [" + mavenCoordinate.getCoordinate() + "] [NONE] ");
                 long date = json.optLong("releaseDate", -1);
-				this.graph = OPALPartialCallGraphConstructor.createPartialJavaCG(mavenCoordinate,
-                        CHA, date, artifactRepository, ONLY_STATIC_CALLSITES);
-                long duration = currentTimeMillis() - startTime; 
+                try {
+                    this.graph = OPALPartialCallGraphConstructor.createPCGFromLocalJar(mavenCoordinate, date,
```

Comment:
Do not use exception handling for programming logic! Concrete proposal: There is no need to have two separate methods, one `createPartialCG` method should be enough, which just takes a path to the jar to be analyzed. The only difference is where the path comes from: either `mavenCoord.getPath()` exists or the .jar is downloaded and a path to the downloaded file is provided. As the jar should *always* exist locally in our setup, I would be fine with just throwing an error when the file is missing.